### PR TITLE
Feature/214/add-firebase-analytics

### PR DESCRIPTION
**Pull request by AmirAgassi**
🔍[Add Firebase Analytics to track how the website is used #214](https://github.com/LaurierHawkHacks/Landing/issues/214)

🔍 **What's Included**
- Implemented Firebase Analytics to gather insights on user interactions within the website.
- In `Analytics.tsx`, initialized Firebase Analytics.
- Updated `SocialIcons.tsx` to log clicks on social media icons, capturing the specific platform clicked.
- Updated `Contact.section.tsx` with analytics event logging for subscription button clicks, tagged with `subscribe_button_click`.
- Modified `Landing.tsx` to track page views on component mount and scroll percentage milestones (every 5%)
    - `first_visit`, `session_start` enabled on GA and handled by default.

🗒️ **Notes**
- Make sure environment variables for Firebase configuration are added in `.env` for testing and deployment.
- We may want to customize our reports on GA to better represent the events collected.

📁 Files Affected:
- `src/components/Analytics.tsx` 
- `src/components/SocialIcons.tsx`
- `src/components/sections/Contact.section.tsx`
- `src/pages/Landing.tsx`

image

### CODE REVIEW

File: src/components/Analytics.tsx
Comment (juancwu) [inline src/components/Analytics.tsx:1]: Not sure if this belongs in the components directory.

Comment (AmirAgassi) [reply src/components/Analytics.tsx:1]: True!!

File: src/utils/Analytics.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+const firebaseConfig = {
+    apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
+    authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
+    databaseURL: process.env.REACT_APP_FIREBASE_DATABASE_URL,
+    projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
+    storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
+    messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
+    appId: process.env.REACT_APP_FIREBASE_APP_ID,
+    measurementId: process.env.REACT_APP_FIREBASE_MEASUREMENT_ID
```
Comment (juancwu) [inline src/utils/Analytics.tsx:12]: We should be using the vite env import way

https://vitejs.dev/guide/env-and-mode.html#env-variables-and-modes

File: src/components/sections/Contact.section.tsx
Comment (juancwu) [inline src/components/sections/Contact.section.tsx:1]: Thinking that we could add a prop to button components for the respective event they are for, this way we can track the buttons in the whole page.

Comment (AmirAgassi) [reply src/components/sections/Contact.section.tsx:1]: Sure thing! That's definitely a better way to do it, we just don't have many buttons on the site and the social ones have their own implementation lol

**Review by juancwu — APPROVED** [review]

**Comment by juancwu** [comment]
@AmirAgassi  only thing left is merge conflicts

**Comment by juancwu** [comment]
Also, don't know if we want to track the clicks on the navbar for each section

**Comment by AmirAgassi** [comment]
> Also, don't know if we want to track the clicks on the navbar for each section

I've implemented it under `navbar_link_click` to be safe, we can always filter the events out of the GA view/reports if we end up not needing it. Lol

**Pull request merged**